lib/active_merchant/billing/gateways/ezidebit.rb

```diff
@@ -80,6 +104,33 @@ def scrub(transcript)
 
       private
 
+      # private
+      # creates a schedule (recurring/subscription plan) for a customer
+      # that was created
+      def create_schedule(money, options = {})
+        request = build_soap_request do |xml|
+          xml['px'].CreateSchedule do
+            add_authentication(xml)
+            xml['px'].EziDebitCustomerID
+            xml['px'].YourSystemReference options[:order_id]
+            xml['px'].ScheduleStartDate options[:start_date]
+            xml['px'].SchedulePeriodType options[:scheduler_period_type]
+            xml['px'].DayOfWeek options[:day_of_week] unless empty?(options[:day_of_week])
+            xml['px'].DayOfMonth options[:day_of_month]
+            xml['px'].FirstWeekOfMonth options[:first_week_of_month] unless empty?(options[:first_week_of_month])
+            xml['px'].SecondWeekOfMonth options[:second_week_of_month] unless empty?(options[:second_week_of_month])
+            xml['px'].ThirdWeekOfMonth options[:third_week_of_month] unless empty?(options[:third_week_of_month])
+            xml['px'].FourthWeekOfMonth options[:fourth_week_of_month] unless empty?(options[:fourth_week_of_month])
+            xml['px'].PaymentAmountInCents amount(money)
+            xml['px'].LimitToNumberOfPayments options[:limit_to_number_of_payments] || 0
+            xml['px'].LimitToTotalAmountInCents options[:limit_to_total_amount_in_cents] || 0
+            xml['px'].KeepManualPayments options[:keep_manual_payments] || 'YES'
```

Comment:
Why in some instances can you use `||` on the options (meaning they’re `nil` and not blank strings), but others you have to run `empty?()`?

Reply:
Because some are mandatory and require a default value, say `0` when no option is set by the caller.

Reply:
and in the other cases, I want to avoid creating the element all together.